### PR TITLE
Add protoype model

### PR DESCRIPTION
**Pull request by meretp**
start prototype for SPDX3.0 
#426

### CODE REVIEW

File: tests/spdx3/model/software/test_file.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import pytest
+
+from spdx3.model.software.software_purpose import SoftwarePurpose
+
+from spdx3.model.software.file import File
+
+
+@mock.patch("spdx3.model.creation_information.CreationInformation", autospec=True)
+def test_correct_initialization(creation_information):
+    file = File("SPDXRef-File", creation_information, content_identifier="https://any.uri",
```
Comment (armintaenzertng) [inline tests/spdx3/model/software/test_file.py:22]: I'm in favor of introducing a fixture module (analogous to the SPDX2 tests) that includes valid defaults for all class initializations. We could then start to use it here, too.

File: tests/spdx3/model/test_integrity_method.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+from spdx3.model.integrity_method import IntegrityMethod
+
+
+def test_correct_initialization():
+    integrity_method = IntegrityMethod(comment="This is a comment.")
```
Comment (armintaenzertng) [inline tests/spdx3/model/test_integrity_method.py:17]: I'm not sure if we need to test initialization of abstract classes

**Review by armintaenzertng — COMMENTED** [review]
Thanks for providing this solid base to work on! :)

File: src/spdx3/model/integrity_method.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+from common.typing.dataclass_with_properties import dataclass_with_properties
+from common.typing.type_checks import check_types_and_set_values
+
+
+@dataclass_with_properties
+class IntegrityMethod:
+    comment: Optional[str] = None
+
+    def __init__(self, comment: Optional[str] = None):
+        check_types_and_set_values(self, locals())
```
[the remark this answers is not in the document]
Comment (meretp) [reply src/spdx3/model/integrity_method.py:22]: Agreed, added comments also to the other abstract classe and openend an issue to investigate if we can implement abstract dataclasses #431

File: tests/spdx3/model/test_creation_information.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# limitations under the License.
+from datetime import datetime
+from unittest import mock
+
+import pytest
+
+from spdx3.model.creation_information import CreationInformation
+
+
+@mock.patch("semantic_version.Version", autospec=True)
```
[the remark this answers is not in the document]
Comment (meretp) [reply tests/spdx3/model/test_creation_information.py:19]: test added

File: tests/spdx3/model/software/test_file.py (hunk 1 above)
Comment (meretp) [reply tests/spdx3/model/software/test_file.py:22]: Agreed, opened an issue for that: #432

File: src/spdx3/model/annotation.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+@dataclass_with_properties
+class Annotation(Element):
+    annotation_type: AnnotationType = None
+    subject: List[Element] = field(default_factory=list)
+    content_type: Optional[str] = None # placeholder for MediaType
+    statement: Optional[str] = None
+    """We overwrite the constructor of the inherited class so that all fields (including the fields from the parent
+    class) are set. Pycharm (and probably also other IDEs) warns about a missing call to the constructor of the super 
+    class but as we have taken care of all fields this warning can be ignored."""
```
[the remark this answers is not in the document]
Comment (meretp) [reply src/spdx3/model/annotation.py:36]: Okay, changed the comment to a one-liner, thanks for the pointer, I didn't know about that.

**Pull request merged**